### ISSUE TITLE
Add custom decorators on top of prisma schema resolvers

### ISSUE THREAD

**Issue by MichalLytek** [MichalLytek/typegraphql-prisma#11]
Is there a way to add custom decorators on top of prisma schema resolvers? e.g. I want for some crud operations to be available to certain roles of users. What is the approach I should use?

_Originally posted by @reflash in https://github.com/MichalLytek/typegraphql-prisma/issues/1#issuecomment-703247069_

How do I add authentication or the `@Authorized` detective for generated resolves? If no is there a workaround?

_Originally posted by @zifahm in https://github.com/MichalLytek/typegraphql-prisma/issues/1#issuecomment-605024859_

Currently securing a graphql api using the first-class [Authorized decorator](https://typegraphql.com/docs/0.17.0/authorization.html#how-to-use) and a custom authenticator hooked into Auth0.

In the current workflow there is a lot of copy-pasting on creation and changes in order to decorate the generated resolvers that are to be secure using the decorator. 
As a feature suggestion, I think it would greatly reduce this manual process by adding a custom Prisma schema decorator on the model level for the CRUD operations, or wherever this would be technically feasible to do.

_Originally posted by @MichaelHindley in https://github.com/MichalLytek/typegraphql-prisma/issues/1#issuecomment-670087746_

**Comment by MichalLytek** [comment]
@reflash 
My proposed solutions is using a config file, similar to https://github.com/maticzav/graphql-shield.

It would allow to apply decorators in a type-safe manner and it could looks something like this:
```ts
import { ResolversEnhanceMap } from "@generated/typegraphql-prisma"`;

const enhanceMap: ResolversEnhanceMap = {
  Query: {
    findManyUsers: [LoggerMiddlware("findManyUsers accessed")],
  },
  Mutation: {
    createUser: [Authorized(Role.ADMIN)],
  },
};

export default enhanceMap;
```

The config object would be imported by generated typegraphql-prisma `index.ts` file and would call the decorators from array manually, like:
`enhanceMap.findManyUsers.forEach(it => it(UserCrudResolver.prototype, "findManyUsers"));`

That would register the decorator on top of a proper class and method, so it will work from TypeGraphQL side like a normal decorator put in the generated class file.

Doing this by runtime file allows to use imports and other language constructs that static prisma attributes (schema docs) are not capable of.

**Comment by reflash** [comment]
@MichalLytek that's great for my use case. one concern I have is that in future it could become too large to maintain as this file will grow with my schema size

**Comment by MichalLytek** [comment]
@reflash 
That config can be defined in a modular fashion - it gonna generate the root tree type composed from smaller subsets, so you can define query in different file, mutations in different, etc. Just 1:1 like with apollo resolvers 😉

**Comment by MichalLytek** [comment]
Closing by 4f93194, available in v0.8.4 🔒
